### PR TITLE
Fix issue #647

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

This PR fixes the performance regression: #647 

### Details and comments

Benchmark result:
main:
```
11.40104109 sec
```
stable/0.3 
```
1.3304943169999999 sec
```
This PR
```
1.3513821619999997 sec
```

### CODE REVIEW

File: qiskit_nature/operators/second_quantization/fermionic_op.py
Comment (mrossinek) [inline qiskit_nature/operators/second_quantization/fermionic_op.py:233]: To be clear: this change is not breaking because the `Sequence` variant was not released yet. So compared to stable the difference is merely adding:
```
            list[tuple[tuple[tuple[str, int], ...], complex]],
```

Comment (mrossinek) [reply qiskit_nature/operators/second_quantization/fermionic_op.py:233]: I just want to check this. Other than that this looks straight forward to me :+1:

Comment (ikkoham) [reply qiskit_nature/operators/second_quantization/fermionic_op.py:233]: This is not breaking change because this is not released.

Comment (ikkoham) [reply qiskit_nature/operators/second_quantization/fermionic_op.py:233]: This is a bit of a discussion, but this part was changed from list to tuple in #607. In that sense, list would be the old API, so should we deprecate it? (In other words, Iterable was introduced as a general expression that included list and tuple, but in real it is not that broad, so I explicitly reverted in this PR.)

Comment (mrossinek) [reply qiskit_nature/operators/second_quantization/fermionic_op.py:233]: What would be the benefit of explicitly deprecated the list-case?

**Review by pbark — APPROVED** [review]
Thanks @ikkoham

File: qiskit_nature/operators/second_quantization/fermionic_op.py
Comment (ikkoham) [reply qiskit_nature/operators/second_quantization/fermionic_op.py:233]: Well, I guess it's nothing special, just a simplification of the code.

Comment (ikkoham) [reply qiskit_nature/operators/second_quantization/fermionic_op.py:233]: Then let's keep it.

**Comment by ikkoham** [comment]
@pbark @mrossinek Thank you for your review. I'll merge this PR after @mrossinek approves.

**Pull request merged**
